### ISSUE TITLE
ERROR: File name too long (os error 36), when rendering with v1.3.242 (v1.3.231 is OK)

### ISSUE THREAD

**Issue by castorfou** [quarto-dev/quarto-cli#4613]
### Bug description

When running action Render and Publish in my blog,

since version 1.3.242 (last week with v1.3.231 it was OK) I have the following error

`ERROR: File name too long (os error 36), stat '/home/runner/work/blog/blog/posts/home/runner/work/blog/blog/data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAbAAAAEeCAYAAADrWEC0AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjMsIGh0dHA6Ly9tYXRwbG90b`

(this is a 70KB base64 png file.

I don't know exactly what happens.

My action file is 

```
on:
  workflow_dispatch:
  push:
    branches: main

name: Quarto Publish

jobs:
  build-deploy:
    runs-on: ubuntu-latest
    permissions:
      contents: write
    steps:
      - name: Check out repository
        uses: actions/checkout@v2

      - name: Set up Quarto
        uses: quarto-dev/quarto-actions/setup@v2
        with:
          version: 'pre-release'

      - name: Render and Publish
        uses: quarto-dev/quarto-actions/publish@v2
        with:
          target: gh-pages
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

I switched to pre-release version to get 1.3 and use code annotation.
Will switch back to stable version for now.

### Checklist

- [ ] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [ ] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [ ] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [ ] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by cscheid** [comment]
Thanks for the report. We need a reproducible .qmd example to be able to track this down.

I _suspect_ the bug on our side is with `self-contained` or `embed-resources`, but we can not make more progress before seeing the actual source of the error.

**Comment by cscheid** [comment]
I think this is a regression in a47b6005604a27ae5cca93e78d40ed2a507ceb61 introduced by the fix to #4544.

**Comment by cscheid** [comment]
@castorfou I tested your blog locally after the fix I pushed to main, and it compiled successfully. (Serious thanks for using github actions, by the way. It made it _really easy_ for us to diagnose and repro!)

We'll build a new release later today which should make your build work.

Thanks for the report again!
